lms/services/exceptions.py

```diff
@@ -238,24 +238,28 @@ class CanvasAPIServerError(CanvasAPIError):
     """
 
 
-class CanvasFileNotFoundInCourse(Exception):
-    """A Canvas file ID wasn't found in the current course."""
-
-    error_code = "canvas_file_not_found_in_course"
-
+class FileNotFoundInCourse(Exception):
```

Comment:
I just realized that having this type here could be useful for course copy (detect a problem in any LMS).

It also avoid a little amount of repetition.

Reply:
Do you think we should be making use of the `SerializableError` here for even less duplication? Something like:

```python
class FileNotFoundInCourse(SerializableError):
    def __init__(self, product_code, file_id):
        super().__init__(
            error_code=f"{{product}}_file_not_found_in_course", 
            details={"file_id": file_id}
        )
```

If we had that, I don't think I'd bother creating a D2L specific sub-class, it could be raised directly with:

```python
raise FileNotFoundInCourse("d2l", file_id)
```

If all the different exceptions are supposed to have the same interface, do we need to have all the different kinds?

Reply:
:+1: Went for basically this approach